sentry-core/src/main/java/io/sentry/core/SentryClient.java

```diff
@@ -150,6 +150,10 @@ private void applyScope(SentryEvent event, @Nullable Scope scope) {
       if (scope.getLevel() != null) {
         event.setLevel(scope.getLevel());
       }
+
+      for (EventProcessor item : scope.getEventProcessors()) {
+        event = item.process(event, hint);
```

Comment:
Where do we apply the event processors from the Options? Maybe we keep them next to each other?
Also, note that we need to bail if a processor returns `null`, instead of passing it to the next processor.
That means the implementor decided to drop the event

Reply:
> Where do we apply the event processors from the Options? Maybe we keep them next to each other?
we do it on `captureEvent` itself, they are in different methods right now because of the `Hint != Cached` check, but they are executed one after the other (EP scopes 1st and EP options 2nd).
> Also, note that we need to bail if a processor returns `null`, instead of passing it to the next processor.
good catch, will do it.
> That means the implementor decided to drop the event